lib/addons/coredns/index.ts

```diff
@@ -0,0 +1,21 @@
+import { CfnAddon } from "@aws-cdk/aws-eks";
+import { ClusterAddOn } from "../..";
+import { ClusterInfo } from "../../stacks/cluster-types";
+
+export class CoreDnsAddOn implements ClusterAddOn {
+
+    version: string;
+
+    constructor(version?: string) {
+        this.version = version ?? "v1.8.0-eksbuild.1";
+    }
+    
+    deploy(clusterInfo: ClusterInfo): void {
```

Comment:
I think we should use a service account for each add-on. @shapirov103 thoughts?

Reply:
Hey @askulkarni2 , other than VPC CNI (for IRSA) where else do you see service accounts will applicable? VPC CNI will allocate enis, so makes sense there.

Reply:
@shapirov103 I don't think the CoreDNS and kube-proxy addons need any AWS permissions. Given that they would get the node role access, should we explicitly attach an IRSA that denies all:all? Not sure about the best practice here.

Reply:
@askulkarni2 I think it is a really good catch for VPC CNI addon - it does require an IRSA setup based on [documentation](https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html). This should be addressed. For the rest of the add-ons, we can skip the SA part since it is not called out in the doc explicitly. 
It could be a security concern, but if we address it with something like deny all (e.g. no permissions role) then we will do it across the board.

Reply:
@AnjanaVaidyanathaIyer this also implies removing the `AmazonEKS_CNI_Policy` from the node role.

Reply:
We discussed this in the morning and agreed to the following scope:

1. If a new cluster provisioned with the managed addon and service account does not add CNI policy to the node role, that will suffice for our case. (pending validation)
2. If the new cluster provisioned with the managed addon and service account adds CNI policy regardless, we will drop the SA support for now and not deal with removal. 
3. There is no support for k8s SA with the managed add-on, they only allow passing a role in with OIDC and trust relationship setup. Because of that we will explore whether creating a fake SA (with OIDC automatically created) will be beneficial for this case. 

Also sounds like a PFR for EKS to deal with policy removal/attachment automatically.

Reply:
HI @shapirov103  and @askulkarni2 , 
I tried to provision a new cluster with managed add-ons and service account. While the stack was launching, I could see that the IAM role attached to the node had CNI policy regardless of the fact that managed add-on for vpc-cni and service account were in place. But the stack failed towards the end with following error.
`Received response status [FAILED] from custom resource. Message returned: Error: b'WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/kubeconfig\nWARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /tmp/kubeconfig\nError: UPGRADE FAILED: another operation (install/upgrade/rollback) is in progress\n' Logs: /aws/lambda/blueprint-construct-dev-awscdkawse-Handler886CB40B-gHnmSjyJ7bQr at invokeUserFunction (/var/task/framework.js:95:19) at processTicksAndRejections (internal/process/task_queues.js:95:5) at async onEvent (/var/task/framework.js:19:27) at async Runtime.handler (/var/task/cfn-response.js:48:13) (RequestId: 3ae7b004-11d1-4b82-95c1-a0182240ca79)`

Seems like a race condition. Could you please help me resolve this issue so that I can do an end to end  test ?

Reply:
Hi @AnjanaVaidyanathaIyer,  I suspect it is not the full stack, there should be some other error more indicative of the issue. 
1. Is it a new stack?
2. Is error happening only when you specify the role arn for the add-on and succeeds otherwise?
3. Is role creation part of the stack and you have a proper dependency for the managed add-on on the role construct?

If all answers are yes, I will vote to move forward with node role (automatic behavior) and create another task to deal with the service account which we will address later.

Reply:
Hi @shapirov103 , 
Yes, it is a new stack. 
1. Error is happening only when I specify the role-arn for vpc-cni add-on. 
2. When I create a new stack with all 3 managed add-ons (without the role-arn for vpc-cni), eks cluster is successfully created.  I can see that the IAM role attached to the node has AmazonEKS_CNI_Policy attached even if it has vpc-cni managed add-on configured during cluster creation. So my best guess is even if I add the role-arn for vpc-cni , node IAM role might still have that policy attached . I will continue testing on why the cluster creation is failing when I try to add the role-arn. I was getting some cdk errors when I was trying to add the dependency. I will give it a go again.